### ISSUE TITLE
A handy way to inject path parameters with the controller base route

### ISSUE THREAD

**Issue by m0uneer** [Romakita/ts-express-decorators#55]
Hello again, guys.

I want a handy way to build the following routes and successfully get the path parameters:

> POST **|** /api/blogs/:blogId/posts/ **|** PostCtrl.addPost()
> POST **|** /api/blogs/:blogId/posts/ **|** PostCtrl.getPosts()
> POST **|**  /api/blogs/:blogId/posts/:postId/comments/ **|** Comments.addComment()
> GET   **|**  /api/blogs/:blogId/posts/:postId/comments/ **|** Comments.getComments()
> ...

What I tried (and I wish it could work) so far:

```javascript
@Controller(`/:postId/comments`)
export class Comments {
  ...
  @Post(`/`)
  public addComment (
    @PathParams('blogId') blogId: string,
    @PathParams('postId') postId: string,
  ) {
    console.log(blogId, postId); // both are undefined
  }
  ...
}

@Controller(`/:blogId/posts`, Comments)
export class PostCtrl {
  ...
}
```

The problem is that the path parameters defined within the controller base route are not injected to the controller api methods. Any ideas?

**Comment by Romakita** [comment]
Hi @m0uneer,

I found this subject http://stackoverflow.com/questions/28977253/express-router-undefined-params-with-router-use-when-split-across-files. 

I add new annotation to configure the Router. So with your example: 
```typescript
@Controller(`/:postId/comments`)
@MergeParams(true)
export class Comments {
  ...
  @Post(`/`)
  public addComment (
    @PathParams('blogId') blogId: string,
    @PathParams('postId') postId: string,
  ) {
    console.log(blogId, postId); // both are undefined
  }
  ...
}

@Controller(`/:blogId/posts`, Comments)
@MergeParams(true)
export class PostCtrl {
  ...
}
```
Add @MergeParams(true) and that will work fine !

Romain

**Comment by Romakita** [comment]
you need to update ts-express-decorators v1.4.5 ;)

**Comment by m0uneer** [comment]
Great, Thanks Romakita :)

**Comment by m0uneer** [comment]
Hi Romatika,
V1.4.8:
Error:(3, 76) TS2305:Module '".../node_modules/ts-express-decorators/lib/index"' has no exported member 'MergeParams'.

**Comment by Romakita** [comment]
Hi @m0uneer ,

Sorry for this bug. It's fixed with v1.4.9.

See you

**Comment by m0uneer** [comment]
Thanks for your efforts.
